### PR TITLE
SALTO-3608: Replace user ids with usernames in Okta

### PR DESCRIPTION
**Pull request by shir-reifenberg**
Replace user ids with usernames in Okta

---

Replace user ids in Okta's instances with user login which is a unique identifier for users in Okta.

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/okta-adapter/src/filters/user.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  [PASSWORD_RULE_TYPE_NAME]: [EXCLUDE_USERS_PATH],
+  [SIGN_ON_RULE_TYPE_NAME]: [EXCLUDE_USERS_PATH],
+  [MFA_RULE_TYPE_NAME]: [EXCLUDE_USERS_PATH],
+}
+
+const isRelevantInstance = (instance: InstanceElement): boolean => (
+  Object.keys(USER_MAPPING).includes(instance.elemID.typeName)
+)
+
+const getUsers = async (paginator: clientUtils.Paginator): Promise<User[]> => {
```
Comment (edenhassid) [inline packages/okta-adapter/src/filters/user.ts:71]: maybe you can cash the results like we do in zendesk

File: packages/okta-adapter/src/filters/user.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+      if (_.isEmpty(relevantChanges)) {
+        return
+      }
+      const users = await getUsers(paginator)
+      if (_.isEmpty(users)) {
+        log.warn('Could not find any users (preDeploy)')
+        return
+      }
+
+      userIdToLogin = Object.fromEntries(
```
Comment (edenhassid) [inline packages/okta-adapter/src/filters/user.ts:159]: can it be different from the mapping in the fetch?

**Review by edenhassid — COMMENTED** [review]
looks good! 😎 added a few comments

File: packages/okta-adapter/src/filters/user.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  const { error } = USERS_RESPONSE_SCHEMA.validate(values)
+  if (error !== undefined) {
+    log.warn(`Received an invalid response for the users: ${error.message}`)
+    return false
+  }
+  return true
+}
+
+const EXCLUDE_USERS_PATH = ['conditions', 'people', 'users', 'exclude']
+const INCLUDE_USES_PATH = ['conditions', 'people', 'users', 'include']
```
[the remark this answers is not in the document]
Comment (shir-reifenberg) [reply packages/okta-adapter/src/filters/user.ts:57]: yes 🥲

File: packages/okta-adapter/src/filters/user.ts (hunk 1 above)
Comment (shir-reifenberg) [reply packages/okta-adapter/src/filters/user.ts:71]: ATM it's not necessary cause we're not calling this function anywhere else

**Pull request merged**